JavaScriptの機能が動かない

# 概要

アプリから JavaScriptを用いた機能が動作していない。
Webブラウザから見た時は正常に動作している。

- スケジュールの一覧で登録状態(緑／赤)が表示されない
- アプリからメンバー用ページ > 個人成績・チーム成績詳細 を見た時データの中身が表示されない

# 原因・対応

アプリ内で保持しているHPのURLが `http://....`で始まっているため CORSの問題が発生し実行されていなかった。
https://github.com/TamaYankees/TamaYenkeesApp/blob/f24d456f32a3bb6638eaa5a65073097c778b6d6c/app/src/main/java/jp/ne/t_yankees/MainActivity.java#L64

Android Studio からAndroidエミュレータでアプリを起動し、PC(Mac)の Chromeから `chrome://inspect/#devices` にアクセスし、エラーの原因を特定  

**以下の 参考）の1~2の手順により判明**

```
Access to XMLHttpRequest at 'https://t-yankees.sakura.ne.jp//api/websb_attendance.cgi' from origin 'http://t-yankees.sakura.ne.jp' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.
schedule_list.js:30 error: {"readyState":0,"status":0,"statusText":"error"}
//api/websb_attendance.cgi:1 Failed to load resource: net::ERR_FAILED
```

----

### 参考）Geminiでデバッグ方法を聞いた結果

AndroidのWebViewでJavaScriptのデバッグ（console.logの確認）を行うには、**PCのChromeのデバック機能（Remote Debugging）**を使うのが最も一般的で確実な方法です。

以下の手順で確認してみてください。
1. WebViewのデバッグを有効にする（アプリ側の修正）

Androidアプリのソースコード（JavaまたはKotlin）で、WebViewのデバッグを許可する設定が必要です。これがないと、PCから接続しても中身が見えません。

```java
// Androidアプリのコード (例: MainActivity.java など)
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
    WebView.setWebContentsDebuggingEnabled(true);
}
```
    注意: セキュリティのため、リリース版（Production）ではこの行を削除するか、BuildConfig.DEBUG で条件分岐させるのが一般的です。

2. PCのChromeからデバッグする
    Android端末をPCにUSB接続します。
    端末の設定で「USBデバッグ」を有効にします。
    PCのChromeブラウザを開き、アドレスバーに以下を入力して移動します。 `chrome://inspect/#devices`
    「Remote Target」にお使いのデバイス名と、実行中のWebViewのタイトル・URLが表示されます。
    「inspect」 をクリックすると、PCと同じデベロッパーツールが立ち上がり、Consoleタブからログを確認できます。

3. WebViewで実行されない際によくある原因
PCでは動くのにWebViewで動かない（doneまで行かない）場合、以下のポイントもチェックしてみてください。
    JavaScriptが無効: WebViewの設定で settings.setJavaScriptEnabled(true); が呼ばれているか確認してください。
    Mixed Content (HTTP/HTTPS混在): APIのURLは https ですが、もし呼び出し元のページが http の場合、セキュリティブロックが発生することがあります。
    CORS (Cross-Origin Resource Sharing): WebViewはブラウザよりもCORSに厳格な場合があります。

    ネットワークセキュリティ設定: Android 9 (API 28) 以上では、デフォルトで暗号化されていない通信（HTTP）をブロックします。今回はHTTPSのようですが、証明書に問題がある場合（オレオレ証明書など）は onReceivedSslError で許可しない限り通信が止まります。
    jQueryの未ロード: $.ajax を使っているため、jQueryがWebView内で正しく読み込まれているか確認してください。

4. 簡易的な確認方法（コード書き換え）
PCに繋げない環境であれば、console.log の代わりに alert() を仕込むか、HTML上のどこかに文字を書き出すように書き換えてデバッグすることも可能です。

```javascript
// エラー内容を画面に表示させる例
}).fail(function(rdata) {
    alert("error: " + JSON.stringify(rdata));
    document.body.innerHTML += "<p>Error: " + JSON.stringify(rdata) + "</p>";
});
```
まずは chrome://inspect/#devices で実際にどのようなエラー（ネットワークエラーなのか、JSの構文エラーなのか）が出ているかを確認することをお勧めします。

アプリ側の `setWebContentsDebuggingEnabled(true)` は設定済みでしょうか？もし未設定であれば、まずはそこから着手してみてください。

--- a/app/src/main/java/jp/ne/t_yankees/MainActivity.java
+++ b/app/src/main/java/jp/ne/t_yankees/MainActivity.java
@@ -61,7 +61,7 @@ public class MainActivity extends AppCompatActivity {
     public static final String EXTRA_POST_DATA = "jp.ne.t_yankees.POST_DATA";
     private static final int REQUEST_CODE_SETTING = 1;
 
-    private final String URL_HP_ROOT = "http://t-yankees.sakura.ne.jp/";
+    private final String URL_HP_ROOT = "https://t-yankees.sakura.ne.jp/";
     private final String WEBSB_CAL = URL_HP_ROOT + "websb3/s-calendar.cgi?";  //スケジュール一覧
     private final String WEBSB_BOARD = URL_HP_ROOT + "websb3/s-webbbs.cgi?";  //メンバー専用掲示板
     private final String WEBSB_SCORE = URL_HP_ROOT + "websb3/s-team.cgi?";  //勝敗結果

